backend/src/bin/web/web_relay.rs

```diff
@@ -1,161 +1,209 @@
+                let msg = msg_get.get_result().unwrap();
+                message = serde_json::to_string(&msg).unwrap();
+            }
+        }
+        Responses::DoorSetState(_) => {
+            if let Responses::DoorSetState(msg_set) = response {
+                assert_eq!(msg_set.get_id().0, id);
+                let msg = msg_set.get_candidate().clone();
+                message = serde_json::to_string(&msg).unwrap();
+            }
+        }
+    }
+    message
 }
 
-
-
-
-// call from client_msg() in main 
+// call from client_msg() in main
 // relays message from web to intercom
-pub async fn listen_for_web(request: WebSocketRequest) -> String{
-
-	let new_request = WebRequests(DeviceCommand(request.command), Message(request.message));
-
-	let request_command = new_request.clone();
-	let command = Commands::match_command(new_request.get_device_command().0);
-	let (setrequest, id): (Requests, u128) = Commands::set_command(&command, request_command);
+pub async fn listen_for_web(request: WebSocketRequest) -> String {
+    let new_request = WebRequests(DeviceCommand(request.command), Message(request.message));
 
-	// send command to intercom and get reply
-	let response = send_command_to_intercom(setrequest).await;
-	let reply_to_web = match_intercom_response(response, id);
+    let request_command = new_request.clone();
+    // let command = Commands::match_command(new_request.get_device_command().0);
+    let command = request_command.get_device_command().0;
+    let (setrequest, id): (Requests, u128) = Commands::set_command(&command, request_command);
 
-	reply_to_web
+    // send command to intercom and get reply
+    let response = send_command_to_intercom(setrequest).await;
+    let reply_to_web = match_intercom_response(response, id);
 
+    reply_to_web
 }
 
-
 async fn send_command_to_intercom(request: Requests) -> Responses {
+    let message = tokio::task::spawn(async move {
+        match TcpStream::connect("127.0.0.1:2000") {
+            Ok(mut intercom_stream) => {
+                // send command to intercom
+                write_to_stream(&mut intercom_stream, &request);
+
+                // get reply
+                let reply: Responses = read_from_stream(&mut intercom_stream).unwrap();
+                reply
+            }
+            Err(e) => {
+                println!("Failed to connect: {}", e);
+                process::exit(1);
+            }
+        }
+    })
+    .await
+    .unwrap();
 
-	let message = tokio::task::spawn(async move {
-		
-		match TcpStream::connect("127.0.0.1:2000") {
-
-			Ok(mut intercom_stream) => {
-				// send command to intercom
-				write_to_stream(&mut intercom_stream, &request);
-				
-				// get reply
-				let reply: Responses = read_from_stream(&mut intercom_stream).unwrap();
-				reply
-			},
-			Err(e) => {
-				println!("Failed to connect: {}", e);
-				process::exit(1);
-			}
-		}
-	}).await.unwrap();
-
-	message
-
+    message
 }
 
-
-
-
-
 // test
```

Comment:
Remove the test functions. Add as integration test the rust way or remove altogether.

Reply:
decided to remove it for now